src/App.tsx

```diff
@@ -1,8 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import * as api from './api';
+import PostsList from './PostList';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadData = async () => {
+    setIsLoading(true);
+
+    const postsFromServer = await api.getPosts();
+
+    setPosts(postsFromServer);
+    setIsLoading(false);
+    setIsLoaded(true);
```

Comment:
You should always use `try...catch` statement when you are using asynchronous, fetch etc.
It helps you identify and handle errors in a controlled manner as far as make debugging much easier.

Here is my suggestion on how to do it.
```suggestion
  const loadPosts = async () => {
    try {
      setIsLoading(true);

      const postsFromServer = await api.getPosts();
      
      setPosts(postsFromServer);
      setIsLoaded(true);
    } catch (error) {
      console.log(error);
    } finally {
      setIsLoading(false);
    }
```

You can read more [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch).